src/main/java/htsjdk/samtools/SAMFileHeader.java

```diff
@@ -324,12 +324,18 @@ public void setAttribute(final String key, final Object value) {
      */
     @Override
     public void setAttribute(final String key, final String value) {
+        String tempVal = value;
         if (key.equals(SORT_ORDER_TAG)) {
             this.sortOrder = null;
+            try {
+                tempVal = SortOrder.valueOf(value).toString();
```

Comment:
In `setSortOrder()` the code uses `.name()` instead of `.toString()` which I think is clearer; either way the code should be consistent.

Also, it seems odd that `setSortOrder()` sets `sortOrder` to non-null, if the argument is a valid `SortOrder`, but this method sets it to `null`. Maybe this method should just call `setSortOrder()` when `key` is `SORT_ORDER_TAG`?

Reply:
@pshapiro4broad that's how the old implementation also does it...it let's the super-class deal with it...take a look at htsjdk.samtools.SAMFileHeader#getSortOrder, I think that might clarify things up (and note that this.sortOrder is private)